src/cdk/a11y/focus-trap/event-listener-inert-strategy.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {FocusTrapInertStrategy} from './focus-trap-inert-strategy';
+import {ConfigurableFocusTrap} from './configurable-focus-trap';
+
+/**
+ * Lightweight FocusTrapInertStrategy that adds a document focus event
+ * listener to redirect focus back inside the FocusTrap.
+ */
+export class EventListenerFocusTrapInertStrategy implements FocusTrapInertStrategy {
+  /** Focus event handler. */
+  private _listener: any;
+
+  /** Adds a document event listener that keeps focus inside the FocusTrap. */
+  preventFocus(focusTrap: ConfigurableFocusTrap): void {
+    this._listener = this._trapFocus.bind(null, focusTrap);
+    focusTrap.ngZone.runOutsideAngular(() => {
+      focusTrap.document.addEventListener('focus', this._listener, true);
+    });
+  }
+
+  /** Removes the event listener added in preventFocus. */
+  allowFocus(focusTrap: ConfigurableFocusTrap): void {
+    focusTrap.document.removeEventListener('focus', this._listener, true);
+    this._listener = null;
+  }
+
+  /**
+   * Refocuses the first element in the FocusTrap if the focus event target was outside
+   * the FocusTrap.
+   */
+  private _trapFocus(focusTrap: ConfigurableFocusTrap, event: FocusEvent) {
+    if (!focusTrap.element.contains(event.target as HTMLElement)) {
+        // Some legacy FocusTrap usages have logic that focuses some element on the page
```

Comment:
Google-only link: https://cs.corp.google.com/piper///depot/google3/cloud/console/web/common/components/subtask/subtask_container.ts?l=128

cdkTrapFocus is enabled the whole time the component is on the page, and right before the component is destroyed, it refocuses the previously focused element, which is outside the FocusTrap.
 
This seems incorrect, since cdkTrapFocus has autoCapture which would handle this correctly, since it refocuses after destroying the FocusTrap. But other internal components have similar behavior (cfc-select also uses cdkTrapFocus, and I think it refocuses the trigger button like this when the select is closed).

Since I want to migrate cdkTrapFocus to use ConfigurableFocusTrap, it seemed easier to support these cases so that we didn't have to fix the failing presubmits one by one.